jobsdb/jobsdb_export.go

```diff
@@ -182,6 +183,9 @@ func (jd *HandleT) getNonMigratedJobsFromDS(ds dataSetT, count int) ([]*JobT, er
 
 	jd.logger.Info(sqlStatement)
 	rows, err = jd.dbHandle.Query(sqlStatement)
+	if err != nil {
+		return nil, fmt.Errorf("query: %w", err)
+	}
 	jd.assertError(err)
```

Comment:
what is the impact of returning an error instead of panicking?
if we should indeed return an error instead of panicking, what's the point of the assertion in line 189?

The above comment stands for the whole method. Do we use the method? Do we have tests for it? Business-wise I cannot judge whether this change is safe or not.

Reply:
I refactor the method to return an error instead of assert/panic. The process that calls this will panic instead with `.Fatalf`.   

`jobsdb_export` is used for scaling out rudder-server instances. Multitenant code will deprecate the jobsdb `export` and `import` needs.


Unfortunately, there are no tests unit or integrations for this functionality. The fact that we are going to deprecate this and the effort required for refactoring/testing make it a low priority.

I can see two options:
  * skip `jobsdb_export`, `jobsdb_import` for deepsource, golangci
  * make those few safe changes to make the linters passing
  
 Correctly understanding, fixing and testing export/import will require a lot of effort, which is better spent on delivering multitenant.